### DISCUSSION TITLE
Support Penpot (eventually)

### DISCUSSION THREAD

**Discussion by jcklpe** [uswds/uswds#4462] [Ideas]
Hi, I wasn't sure if this had been brought attention to the USWDS team but I wanted to recommend ya'll check out the new open source Figma alternative: Penpot

https://penpot.app/

https://github.com/penpot/penpot

Penpot is a wireframing app. It's browser and cloud software, similar to Figma. [They don't yet support component variants](https://github.com/penpot/penpot/issues/768) (as in Figma) or states (as in XD), but they have made a lot of progress and it's by far the best open source wireframing tool I've seen. 

Reasons why USWDS should prioritize supporting Penpot:

- It's open source
- no vendor lock-in
- It's free to self-host
- cross platform (unlike Sketch)

Even though I don't think Penpot is quite to the level needed for USWDS to putting their support behind, I think it would be a good idea to get that idea under consideration sooner rather than later. This seems like a no brainer to me. Government work benefits from open source and being open sourced, and from being built with open source tools.

**Comment by mejiaj** [comment]
Thanks for the suggestion! The Figma variant is thanks to a community effort, we mention that here in [Getting Started for Designers](https://designsystem.digital.gov/documentation/getting-started-for-designers/). 

If you create a penpot variant we'd be happy to add it as a community resource.

**Reply by jcklpe** [reply]
I'll make one as soon as the get the Variant feature added, for sure :D  

I'm actually doing a rebuild of the Figma ui kit that trussworks put out. As far as I can tell it's just a programmatic import of the official sketch ui kit and it's... (no offense to trussworks) a little lacking. No auto layout, variants are all screwed up, component sets are wonky and disorganized. Feel free to check it out here: https://www.figma.com/file/A0yoxLJsUfS1n2TVcPAyR0/USWDS-2.12.0?node-id=0%3A1

**Reply by jcklpe** [reply]
it's still a work in progress. I've just been cleaning stuff up on the weekends. Stills lots of nasty unnamed layers to label and clean up. But it's getting there.

**Comment by mgifford** [comment]
I like this idea, but wondering if we could use it to get more than we can via Figma. Because it is open source, perhaps there is a way to help keep it in better sync with the HTML output. Perhaps we could add more accessibility validation as part of a Penpot implementation of the design system.  


What would help us innovate and more easily?

**Reply by jcklpe** [reply]
The self hosting would also allow it to get internal use in orgs like DHS that have policies against offsite server hosting I believe.

**Comment by mpaiva** [comment]
Would now be a good time to revive this discussion? @jcklpe and @mgifford would you be interested in having a discussion on how we might be able to collaborate and create a set of USWDS library and templates in Penpot? 

I can also try bring Penpot (Pablo Ruiz-Múzquiz) and the Tokens Studio team (Jan Six and Marco Krenn) into this thread, as I assume they'd be very much interested in outreaching their efforts to the USWDS community.

**Reply by jcklpe** [reply]
@mpaiva  Yes, I'd be very interested in helping if I could. 

I think the biggest feature still missing from Penpot is component variants, but other than that I think it's quite robust, and I'm def interested in learning more about the new token feature set.

**Reply by mpaiva** [reply]
Adding @mzebley here, he is our USWDS design token wizard

**Reply by jcklpe** [reply]
Wanted to check in on this? If ya'll would like I can hop on a call to chat. Calendly link for me here: schedule.aslanfrench.work
